ironfish/src/consensus/verifier.ts

```diff
@@ -124,6 +124,11 @@ export class Verifier {
       return { valid: false, reason: VerificationResultReason.INVALID_MINERS_FEE }
     }
 
+    const expectedTransactionCommitment = transactionCommitment(block.transactions)
+    if (!expectedTransactionCommitment.equals(block.header.transactionCommitment)) {
+      return { valid: false, reason: VerificationResultReason.INVALID_TRANSACTION_COMMITMENT }
+    }
```

Comment:
We should do the transaction commitment check before the other transaction-related checks in this function, so that we're sure the transactions are expected for the block before we verify them.

I think in the addBlock process we should also say that if a block has an invalid transaction commitment, we don't mark the block's hash as invalid. We can't know if a peer sent the right transactions for the block or not. That could go in a different PR though

Reply:
Yea you're right. I was thinking the other way around but we don't want to waste time verifying the transactions